src/org/labkey/test/pages/study/CreateNewVisitPage.java

```diff
@@ -0,0 +1,98 @@
+package org.labkey.test.pages.study;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.components.html.Checkbox;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.components.html.SelectWrapper;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+
+public class CreateNewVisitPage extends LabKeyPage<CreateNewVisitPage.ElementCache>
+{
+    public CreateNewVisitPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static CreateNewVisitPage beginAt(WebDriverWrapper webDriverWrapper)
+    {
+        return beginAt(webDriverWrapper, webDriverWrapper.getCurrentContainerPath());
+    }
+
+    public static CreateNewVisitPage beginAt(WebDriverWrapper webDriverWrapper, String containerPath)
+    {
+        webDriverWrapper.beginAt(WebTestHelper.buildURL("study", containerPath, "createVisit"));
+        return new CreateNewVisitPage(webDriverWrapper.getDriver());
+    }
+
+    public CreateNewVisitPage setLabel(String label)
+    {
+        elementCache().label.set(label);
+        return this;
+    }
+
+    public CreateNewVisitPage setMinSequence(String range)
+    {
+        elementCache().sequenceNumMin.set(range);
+        return this;
+    }
+
+    public CreateNewVisitPage setMaxSequence(String range)
+    {
+        elementCache().sequenceNumMax.set(range);
+        return this;
+    }
+
+    public CreateNewVisitPage setDescription(String value)
+    {
+        setFormElement(elementCache().description, value);
+        return this;
+    }
+
+    public CreateNewVisitPage setType(String option)
+    {
+        elementCache().type.selectByVisibleText(option);
+        return this;
+    }
+
+    public CreateNewVisitPage setVisitHandling(String value)
+    {
+        elementCache().sequenceNumHandling.selectByVisibleText(value);
+        return this;
+    }
+
+    public CreateNewVisitPage setShowByDefault(boolean value)
+    {
+        elementCache().showByDefault.set(value);
+        return this;
+    }
+
+    public ManageVisitPage clickSave()
+    {
+        clickAndWait(elementCache().save);
+        return new ManageVisitPage(getDriver());
+    }
+
```

Comment:
Do you have / need a test that clicks the Cancel button?
Is there a possibility that clicking Save could throw an error? Like a required field not set, or a data collision with existing data (e.g. duplicate labels, `sequenceNumMin` is greater than `sequenceNumMax`, overlapping ranges, etc...). Then you would need a `clickSaveExpectingError` method and the appropriate test cases.
You may want to sync with @labkey-klum on some of those.

Reply:
This is very old page, i created a page bare minimum methods that i needed for testing new feature